lara-typescript/src/interactive-api-client/index.ts

```diff
@@ -0,0 +1,344 @@
+              if (this.options.onInitInteractive) {
+                this.log("debug", "initInteractive listener called, calling client callback");
+                this.options.onInitInteractive(initMessage);
+              } else {
+                this.log("debug", "initInteractive listener called, no client callback found");
+              }
+          });
+
+          this.addListener("loadInteractive", (interactiveState: InteractiveState | null) => {
+            if (this.options.onLoadInteractive) {
+              this.log("debug", "loadInteractive listener called, calling client callback");
+              this.options.onLoadInteractive(interactiveState);
+            } else {
+              this.log("debug", "loadInteractive listener called, no client callback found");
+            }
+          });
+
+          this.phone.initialize();
+        } else {
+          this.log("info", "#connect: this.phone already connected");
+        }
+        return true;
+      }
+      this.log("error", "#connect: not in iframe");
+      return false;
+    }
+
+    public disconnect() {
+      if (inIframe) {
+        if (this.phone) {
+          this.log("debug", "#disconnect: disconnecting");
+          this.phone.disconnect();
+          this.phone = undefined;
+        } else {
+          this.log("info", "#disconnect: this.phone already disconnected");
+        }
+        return true;
+      }
+      this.log("error", "#disconnect: not in iframe");
+      return false;
+    }
+
+    public get inIFrame() {
+      return inIframe;
+    }
+
+    public setLearnerUrl(url: string | null) {
+      return this.post("setLearnerUrl", url);
+    }
+
+    public setInteractiveState(interactiveState: InteractiveState | string | null) {
+      return this.post("interactiveState", interactiveState);
+    }
+
+    public setHeight(height: number | string) {
+      return this.post("height", height);
+    }
+
+    public setSupportedFeatures(supportedFeatures: ISupportedFeatures) {
+      return this.post("supportedFeatures", supportedFeatures);
+    }
+
+    public setNavigation(options: INavigationOptions) {
+      return this.post("navigation", options);
+    }
+
+    public setAuthoredState(authoredState: AuthoredState) {
+      return this.post("authoredState", authoredState);
+    }
+
+    public getAuthInfo(): Promise<IAuthInfo> {
+      return new Promise<IAuthInfo>((resolve, reject) => {
+        if (!this.phone) {
+          this.log("debug", "#getAuthInfo: this.phone undefined");
+          return reject("Not in iframe");
+        }
+        const listener = (authInfo: IAuthInfo) => {
+          resolve(authInfo);
+          this.removeListener("authInfo");
+        };
+        this.addListener("authInfo", listener);
+        this.post("getAuthInfo");
+      });
+    }
+
+    public getFirebaseJWT(options: IGetFirebaseJwtOptions): Promise<string> {
+      return new Promise<string>((resolve, reject) => {
```

Comment:
This pattern matches the one above, so just like that one I think this should have an id added to the messages.